### ISSUE TITLE
VariableSizeList can call itemSize getter with an invalid index

### ISSUE THREAD

**Issue by chrisnojima** [bvaughn/react-window#62]
Hey,

I recently ported a list over to react-window and I noticed that under certain circumstances the itemSize callback can be called with an invalid index. 

My code is basically

```
                <VariableSizeList
                  height={height}
                  width={width}
                  ref={this._setRef}
                  onItemsRendered={this._onItemsRendered}
                  itemCount={this.props.rows.length}
                  itemSize={this._itemSizeGetter}
                  estimatedItemSize={56}
                >
...
  _itemSizeGetter = index => {
    if (this.props.filter.length) {
      return 56
    }
    const row = this.props.rows[index]

```
so itemCount is based on this.props.rows so that should always be synced up.

From what I've seen itemCount is `0` and since (https://github.com/bvaughn/react-window/blob/master/src/VariableSizeList.js#L42)
```
    for (let i = lastMeasuredIndex + 1; i <= index; i++) {
      let size = ((itemSize: any): itemSizeGetter)(i);
```
the `i <= index` allows this to be out of bounds

**Comment by bvaughn** [comment]
Hi. Can you provide a full repro case please? Either a test or a Code Sandbox would do. This issue doesn't have enough info to be actionable as is.

**Comment by chrisnojima** [comment]
sure: https://codesandbox.io/s/rmoj05q95m

**Comment by bvaughn** [comment]
Ah, so the specific problem is when you pass an `itemCount` of zero.

Fixed in c560b18. Will be released in a bugfix shortly.

**Comment by bvaughn** [comment]
Published in [1.2.1](https://github.com/bvaughn/react-window/blob/master/CHANGELOG.md#121)
